[#1979] Change date captions for cases list as design

jiromaykin opened the pull request:
issue https://taiga.maykinmedia.nl/project/open-inwoner/task/1979 
Dates in case list were not put in the positions they needed to be, so now put conforming to design (closely aligned to the Case title) 
And: also improved some of the Card paddings according to design + changed the name of the gray variant to a "contextual" name that is more commonly used by designers and with design-tokens ('900' instead of '90').

Link to design: https://www.figma.com/file/iKGhWhstaLIlFSaND2q7cE/OIP---Designs-(new)?type=design&node-id=1%3A273&mode=design&t=PfzsswiRMAp5biaL-1

pi-sigma commented on src/open_inwoner/cms/cases/tests/test_htmx.py:
```diff
@@ -423,7 +423,7 @@ def test_cases(self, m):
 
         # case title
-        case_title = page.get_by_role("link", name=self.zaaktype["omschrijving"])
+        case_title = page.get_by_text("Coffee zaaktype")
```
This is not right, given that `self.zaaktype["omschrijving"]` = `"Coffee zaaktype"`.

The problem is `aria-label="{% trans "Bekijk aanvraag" %}"` in the link of the case. The accessible name for the link is given by the content of the `h4` element, hence Playwright gets confused and thinks the `name` for the `link` role is "Bekijk aanvrag", the content of `aria-label`. If you replace `aria-label` with `aria-labelledby` and link it to the `h4`, the test works. Two more comments:
- Strictly speaking, the content of the link is much more than "Coffee zaaktype"; it contains the information of all the list elements inside `<a>....</a>`. I don't know if it's still appropriate to use `aria-labelledby` in this case, linking it to only part of the content, though my intuition is that it's OK.
- If you change the html as suggested, you should also change it for the open submission, which has the same structure. We don't have a separate test for those, but the situation is the same and should be handled accordingly.

jiromaykin replied:
Ah yes, I actually need to rethink this. An aria-label is not needed in case of an anchor-surrounding/overriding-everything but we probably still want a separate label for the button inside the anchor.

jiromaykin replied:
@pi-sigma After deeper consideration: I've removed the title attributes since those are actually very bad for accessibility - also the aria-labels are not necessary here because the screenreaders have access to all texts and can read everything (except for the purposefully hidden icons).
(I should start removing unnecessary labels throughout the entire project).
If the component follows correct HTML you don't need aria-labels, unless you wish to override something.

pi-sigma commented on src/open_inwoner/templates/pages/cases/list_inner.html:
```diff
@@ -16,38 +16,41 @@ <h1 class="utrecht-heading-1" id="cases">{{ page_title }} ({{ paginator.count }}
                 {% if case.case_type == "OpenSubmission" %}
-                    <a href="{{ case.vervolg_link }}"  class="cases__link">
-                    <p class="h4"><span class="link link__text">{{ case.naam }}</span></p>
+                    <a href="{{ case.vervolg_link }}" class="cases__link" aria-label="{% trans "Bekijk aanvraag" %}" title="{% trans "Bekijk aanvraag" %}">
```
Since you changed the `span` element below to "Voltooi aanvraag" (which is a nice way to capture the distinction between cases and open submissions), you should change this too. Also, see previous comment about `aria-label` and `aria-labelledby`.

pi-sigma commented on src/open_inwoner/templates/pages/cases/list_inner.html:
```diff
@@ -16,38 +16,41 @@ <h1 class="utrecht-heading-1" id="cases">{{ page_title }} ({{ paginator.count }}
                 {% else %}
-                    <a href="{% url 'cases:case_detail' object_id=case.uuid %}" class="cases__link">
-                    <p class="h4"><span class="link link__text">{{ case.description }}</span></p>
+                    <a href="{% url 'cases:case_detail' object_id=case.uuid %}" class="cases__link" aria-label="{% trans "Bekijk aanvraag" %}" title="{% trans "Bekijk aanvraag" %}">
```
This is the `aria-label` that causes the problem.

pi-sigma commented on src/open_inwoner/templates/pages/cases/list_inner.html:
```diff
@@ -16,38 +16,42 @@ <h1 class="utrecht-heading-1" id="cases">{{ page_title }} ({{ paginator.count }}
-                    <a href="{{ case.vervolg_link }}"  class="cases__link">
-                    <p class="h4"><span class="link link__text">{{ case.naam }}</span></p>
+                    <a href="{{ case.vervolg_link }}" class="cases__link">
+                    <h2 class="card__heading--h4"><span class="link link__text">{{ case.naam }}</span></h2>
```
Maybe remove the `h4` from the class and just use a descriptive class name: `car__heading`, `card__date-caption`, or sth. like this.

jiromaykin replied:
@pi-sigma Yes, it looks like in the new design ALL headings are going to be 18px, so I'll need to make this a generic heading class, mostly set on H2.

pi-sigma reviewed (commented):
Now that the date has been moved closer to the title, the distance between two lines of a multi-line title is bigger than the distance between title and date. Should we change this?

![Screenshot from 2024-01-30 10-33-37](https://github.com/maykinmedia/open-inwoner/assets/89185369/c67b712c-c6ac-456b-bc16-bbfcfc454457)

jiromaykin commented:
> Now that the date has been moved closer to the title, the distance between two lines of a multi-line title is bigger...

I have changed the line-height and changed the default text for open submissions button.
